src/clr-angular/forms/common/layout.ts

```diff
@@ -0,0 +1,27 @@
+/**
+ * Copyright (c) 2016-2018 VMware, Inc. All Rights Reserved.
+ * This software is released under MIT license.
+ * The full license information can be found in LICENSE in the root directory of this project.
+ */
+
+import { Directive, ElementRef, Input, Renderer2 } from '@angular/core';
+import { Layouts, LayoutService } from './providers/layout.service';
+
+@Directive({
+  selector: '[clrLayout]',
+})
+export class ClrLayout {
+  @Input('clrLayout') layout: Layouts;
+
+  constructor(private el: ElementRef, private renderer: Renderer2, private layoutService: LayoutService) {}
+
+  ngAfterContentInit() {
+    this.renderer.addClass(this.el.nativeElement, 'clr-form-' + this.layoutService.layout);
```

Comment:
I am not really sure why this line of code should use `Renderer2` and placed inside `ngAfterContentInit`. 

Also, I am just not really fond of accessing a part of a class name from another service and concatenating here. Can we access the entire class name from that same service? In that case, if this class name changes, we would need to make changes to only that service file.

Reply:
To do this you have to know the host property value at compile time. Like this:

```
host: {
  '[class.clr-form-{layoutType}]: 'true'
}
```

As far as the class name, I could have the service generate it instead of here.

Reply:
We can do this:
```html
host: {
    "[class]": "layoutTypeClass",
 }
```
```ts
this.layoutTypeClass = 'clr-form-' + this.layoutService.layout;
```
With this, we don't even need to inject `Renderer2` and `ngAfterContentInit` hook